python/pyiceberg/table/__init__.py

```diff
@@ -191,11 +445,14 @@ def from_metadata(cls, metadata_location: str, properties: Properties = EMPTY_DI
 
         metadata = FromInputFile.table_metadata(file)
 
+        from pyiceberg.catalog.null import NoopCatalog
+
         return cls(
             identifier=("static-table", metadata_location),
             metadata_location=metadata_location,
             metadata=metadata,
             io=load_file_io({**properties, **metadata.properties}),
+            catalog=NoopCatalog("static-table"),
```

Comment:
Couldn't we just check that the catalog is `None` and throw an error? Seems like we don't necessarily need the `NoopCatalog`.

Reply:
I'd rather have the `NoopCatalog` because otherwise, we need to not-None check everywhere we use the catalog.